compiler/bindings/python/iree/compiler/tools/import_onnx/importer_externalization_overrides.py

```diff
@@ -187,32 +221,53 @@ def define_function(
             context_cache=cc,
             module_op=module_op,
             module_cache=mc,
-            num_elements_threshold=num_elements_threshold,
-            params_scope=params_scope,
+            param_data=param_data,
         )
         for node_name, input_value in zip(graph_info.input_map.keys(), block.arguments):
             imp._nv_map[node_name] = input_value
         imp._populate_graph_attrs(func_op)
+        imp._gi.input_map = all_input_map
         return imp
 
+    def import_all(self, func=True):
+        num_inputs = len(self._gi.input_map.items())
+        if (
+            self.param_data.input_index_threshold is not None
+            and self.param_data.input_index_threshold < num_inputs
+        ):
+            thresh = (
+                self.param_data.input_index_threshold
+                if self.param_data.input_index_threshold >= 0
+                else max(0, self.param_data.input_index_threshold + num_inputs)
+            )
+            for i in range(thresh, num_inputs):
+                self.import_initializer(list(self._gi.input_map.values())[i])
+        super().import_all(func)
+
     def import_initializer(
-        self, initializer: onnx.TensorProto, extern_name: Optional[str] = None
+        self,
+        initializer: Union[onnx.TensorProto, onnx.ValueInfoProto],
+        extern_name: Optional[str] = None,
     ) -> Value:
         # If an explicitly specified name is given, use that; otherwise, pick
         # up the name from the tensor proto itself
         initializer_name = extern_name if extern_name else initializer.name
-        dims = list(initializer.dims)
-        num_elements = 1
-        for d in dims:
-            num_elements = num_elements * d
-        if num_elements < self.num_elements_threshold:
-            imported_tensor = super().import_initializer(initializer)
-            self._nv_map[initializer_name] = imported_tensor
-            return imported_tensor
+        if isinstance(initializer, onnx.TensorProto):
+            dims = tuple(initializer.dims)
+            num_elements = 1
+            for d in dims:
+                num_elements = num_elements * d
+            if num_elements < self.param_data.num_elements_threshold:
+                imported_tensor = super().import_initializer(initializer)
+                self._nv_map[initializer_name] = imported_tensor
+                return imported_tensor
+            data_type = initializer.data_type
+        else:
+            dims, data_type = self.get_type_info_from_type(initializer.type)
```

Comment:
Might as well check that `initializer` is an instance of `onnx.ValueInfoProto` and raise an exception if an unexpected type was passed:
```suggestion
        if isinstance(initializer, onnx.TensorProto):
            dims = tuple(initializer.dims)
            num_elements = 1
            for d in dims:
                num_elements = num_elements * d
            if num_elements < self.param_data.num_elements_threshold:
                imported_tensor = super().import_initializer(initializer)
                self._nv_map[initializer_name] = imported_tensor
                return imported_tensor
            data_type = initializer.data_type
        elif isinstance(initializer, onnx.ValueInfoProto):
            dims, data_type = self.get_type_info_from_type(initializer.type)
        else:
            raise TypeError(f"Expected an onnx.TensorProto or onnx.ValueInfoProto, received '{type(initializer)}'")
```